tests/test_categories.py

```diff
@@ -0,0 +1,109 @@
+import pytest
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.testing import mock
+from starlette import status
+from starlette.datastructures import ImmutableMultiDict
+
+from app.database.models import Event
+from app.routers.categories import get_user_categories, validate_request_params
+
+
+class TestCategories:
+    CATEGORY_ALREADY_EXISTS_MSG = "category is already exists for"
+    UNALLOWED_PARAMS = "contains unallowed params"
+
+    @staticmethod
+    def test_get_categories_logic_succeeded(session, user, category):
+        assert get_user_categories(session, category.user_id) == [category]
+
+    @staticmethod
+    def test_creating_new_category(client, user):
+        response = client.post("/categories/",
+                               json={"user_id": user.id, "name": "Foo",
+                                     "color": "eecc11"})
+        assert response.ok
+        assert {"user_id": user.id, "name": "Foo",
```

Comment:
I think it's better to leave out the `items()` call. The `json` might not be in the same order

Reply:
changed to set, so now order is not an issue ;)